### PR TITLE
kodiPlugins: import addons from kodi repositories

### PR DESCRIPTION
**Pull request by aanderse**
###### Motivation for this change
It is difficult to package `kodi` plugins because of the dependencies that most come with. This PR aims to improve the situation by automatically importing every plugin from the `kodi` repository that we reasonably can automatically import.

I'm not sure if this PR is a good idea or not:
- this adds over 1,000 packages to `nixpkgs`
- not all `kodi` plugins can be packaged for `nixpkgs` because some plugins require read/write access to other plugins source code at runtime... see `kodiPlugins.skin-arctic-zephyr-mod` which I have packages as an example as such (hint: the plugin crashes on startup)
- plenty of other reasons :man_shrugging: 

On the plus side this allows you to create an instance of `kodi` with plugins that won't get mangled by other `kodi` plugins, and can avoid the headache of having to manually create custom `kodi` builds, which I'm in the habit of doing every 6 months or so. In other words this PR has high value to me... but maybe only me :man_shrugging: 

**This PR is heavily work in progress and I'm looking for feedback on whether this even belongs in `nixpkgs` or not**

ping @spiderbit @tyrion

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
> * this adds over 1,000 packages to `nixpkgs`

Would it be a smarter idea to manually add and verify the plugins people want to use?

File: pkgs/applications/video/kodi/plugins.nix [hunk 1]
```diff
@@ -556,4 +556,107 @@ let self = rec {
 
     extraBuildInputs = [ libarchive lzma bzip2 zlib lz4 lzo openssl ];
   };
-}; in self
+
+  # an example of a plugin that won't work when packaged with nix
+  # because other packages will modify this package at runtime...
+  skin-arctic-zephyr-mod = mkKodiPlugin rec {
+    plugin = "skin-artic-zephyr-mod";
+    namespace = "skin.arctic.zephyr.mod";
```
Comment (SuperSandro2000) [inline pkgs/applications/video/kodi/plugins.nix:564]: Can we generate one name of the other?

Comment (aanderse) [reply pkgs/applications/video/kodi/plugins.nix:564]: I agree, we should be doing that. Unfortunately not all addons are named according to this yet... but we can fix that.

**Comment by aanderse** [comment]
Yeah I was considering selectively adding plugins as needed... it might be challenging to teach the @r-ryantm bot to update them, though (I don't know `haskell` to add the functionality).

File: pkgs/applications/video/kodi/plugins.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/video/kodi/plugins.nix:564]: Maybe default it to this and overwrite it where necessary?

**Comment by SuperSandro2000** [comment]
> Yeah I was considering selectively adding plugins as needed... it might be challenging to teach the @r-ryantm bot to update them, though (I don't know `haskell` to add the functionality).

r-ryantm respects update scripts so if you can write a script in bash, python or maybe perl the bot can update it. There is an update script in the maintainers directory which I think the bot uses.

**Comment by aanderse** [comment]
> r-ryantm respects update scripts so if you can write a script in bash, python or maybe perl the bot can update it. There is an update script in the maintainers directory which I think the bot uses.

I didn't know that. I'll give my approach some thought. Thanks.

**Pull request closed without merging**